packages/ui/src/components/busSelector.tsx

```diff
@@ -75,6 +75,7 @@ const BusSelector = memo(function BusSelector({ agencies }: BusSelectorProps) {
       enabled: Boolean(agency) && Boolean(routeName),
       onSuccess(data) {
         dispatch({ type: 'route', value: data })
+        vehiclesDispatch({ type: 'set', value: undefined })
```

Comment:
Otherwise the map's vehicle layer would keep adding markers as users changed routes within the same agency, i.e. this has the side effect of clearing the map's vehicle marker layer.